### ISSUE TITLE
Cannot determine type of a function with a no_type_check decorator

### ISSUE THREAD

**Issue by JukkaL** [JukkaL/mypy#663]
Mypy complains if I call a `no_type_check` function above the definition:

```
import typing

def f() -> None:
    foo()  # Error: Cannot determine type of 'foo'

@typing.no_type_check
def foo(x: {1:2}) -> [1]:
    1 + 'x'
```

We should detect `no_type_check` during semantic analysis and set the type of the `Decorator` object to `Any` there, before type checking.
